services/galaxytoolshed/galaxy-toolshed-downloads.service.js

```diff
@@ -0,0 +1,55 @@
+import Joi from 'joi'
+import { renderDownloadsBadge } from '../downloads.js'
+import BaseGalaxyToolshedService from './galaxy-toolshed-base.js'
+
+const repositoryRevisionInstallInfoSchema = Joi.array().items(
+  Joi.object({
+    times_downloaded: Joi.number().required(),
+  }).required(),
+  Joi.object({}),
+  Joi.object({})
+)
+
+export default class GalaxyToolshedDownloads extends BaseGalaxyToolshedService {
+  static category = 'downloads'
+  static route = {
+    base: 'galaxy-toolshed/downloads',
+    pattern: ':repository/:owner',
+  }
+
+  static examples = [
+    {
+      title: 'Galaxy Toolshed - Downloads',
+      namedParams: {
+        repository: 'sra_tools',
+        owner: 'iuc',
+      },
+      staticPreview: this.render({ downloads: 10000 }),
+    },
+  ]
+
+  static defaultBadgeData = {
+    label: 'downloads',
+  }
+
+  static render({ downloads }) {
+    return renderDownloadsBadge({ downloads })
+  }
+
+  static transform(response) {
+    const data = this.filterRepositoryRevisionInstallInfo({
+      response,
+    })
+    return data.times_downloaded
+  }
+
```

Comment:
Not strictly necessary, but we can get rid of this too since we just need to grab one field from a fixed, well known position in the response. I'll provide the specifics down below in the `handle` function but if you prefer you could keep this `transform` function and replace this function's body with the suggested line below for extracting the download count

```suggestion
```

Reply:
Updated into 238d9607c